src/vmm/src/device_manager/mmio.rs

```diff
@@ -102,39 +103,30 @@ pub struct MMIODeviceInfo {
 /// Manages the complexities of registering a MMIO device.
 pub struct MMIODeviceManager {
     pub(crate) bus: devices::Bus,
-    pub(crate) irq_allocator: IdAllocator,
-    pub(crate) address_allocator: AddressAllocator,
     pub(crate) id_to_dev_info: HashMap<(DeviceType, String), MMIODeviceInfo>,
 }
 
 impl MMIODeviceManager {
     /// Create a new DeviceManager handling mmio devices (virtio net, block).
-    pub fn new(
-        mmio_base: u64,
-        mmio_size: u64,
-        (irq_start, irq_end): (u32, u32),
-    ) -> Result<MMIODeviceManager> {
+    pub fn new() -> Result<MMIODeviceManager> {
         Ok(MMIODeviceManager {
-            irq_allocator: IdAllocator::new(irq_start, irq_end).map_err(Error::AllocatorError)?,
-            address_allocator: AddressAllocator::new(mmio_base, mmio_size)
-                .map_err(Error::AllocatorError)?,
             bus: devices::Bus::new(),
             id_to_dev_info: HashMap::new(),
         })
     }
 
     /// Allocates resources for a new device to be added.
-    fn allocate_mmio_resources(&mut self, irq_count: u32) -> Result<MMIODeviceInfo> {
-        let irqs = (0..irq_count)
-            .map(|_| self.irq_allocator.allocate_id())
-            .collect::<vm_allocator::Result<_>>()
+    pub(crate) fn allocate_mmio_resources(
```

Comment:
```suggestion
    fn allocate_mmio_resources(
```
As per review on #3124 , we don't need to make this public.
This associated function is used only inside this file.
Keeping this private can avoid misuse (e.g., only allocates MMIO resources, but not register them to KVM).

Reply:
You are right! Ty!